### PR TITLE
[Feat] Sidechain Pausable

### PR DESCRIPTION
**Pull request by phijfry**

### CODE REVIEW

File: contracts/sidechain/AuraProxyOFT.sol [hunk 1]
```diff
@@ -45,8 +48,16 @@ contract AuraProxyOFT is ProxyOFT {
      * @param _sender Address that is locking
      * @param _amount Amount to lock
      */
-    function _lockFor(address _sender, uint256 _amount) internal {
-        IAuraLocker(locker).lock(_sender, _amount);
+    function _lockFor(
+        uint16 _srcChainId,
```
Comment (0xahtle7) [inline contracts/sidechain/AuraProxyOFT.sol:52]: Nice `uint16 _srcChainId,`

File: contracts/sidechain/PausableOFT.sol [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { OFT } from "../layerzero/token/oft/OFT.sol";
+import { PauseGaurdian } from "./PauseGuardian.sol";
+
+/**
+ * @title AuraOFT
```
Comment (0xahtle7) [inline contracts/sidechain/PausableOFT.sol:8]:
```suggestion 
 * @title PausableOFT
```

File: contracts/sidechain/PausableProxyOFT.sol [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+     * @param _amount       Amount to transfer
+     * @param _timestamp    Time when this transfer was queued
+     */
+    function processQueued(
+        uint256 _epoch,
+        uint16 _srcChainId,
+        address _to,
+        uint256 _amount,
+        uint256 _timestamp
+    ) external whenNotPaused {
```
Comment (0xahtle7) [inline contracts/sidechain/PausableProxyOFT.sol:163]: who should be able to call this `processQueued` ? anyone?

File: contracts/sidechain/PausableProxyOFT.sol [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        bytes32 queueRoot = keccak256(abi.encode(_epoch, _srcChainId, _to, _amount, _timestamp));
+        require(queue[queueRoot], "!root");
+        require(_timestamp.add(queueDelay) < block.timestamp, "!timestamp");
```
Comment (0xahtle7) [inline contracts/sidechain/PausableProxyOFT.sol:166]:
```suggestion
        require(_timestamp.add(queueDelay) < block.timestamp, "!delayedTime");
```

File: contracts/sidechain/PauseGuardian.sol [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @title PauseGaurdian
+ */
+contract PauseGaurdian is Pausable {
+    /* -------------------------------------------------------------------
+       Storage 
+    ------------------------------------------------------------------- */
+
+    address public immutable gaurdian;
```
Comment (0xahtle7) [inline contracts/sidechain/PauseGuardian.sol:14]:
```suggestion
guardian
```

File: contracts/sidechain/PausableProxyOFT.sol [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+     */
+    function processQueued(
+        uint256 _epoch,
+        uint16 _srcChainId,
+        address _to,
+        uint256 _amount,
+        uint256 _timestamp
+    ) external whenNotPaused {
+        bytes32 queueRoot = keccak256(abi.encode(_epoch, _srcChainId, _to, _amount, _timestamp));
+        require(queue[queueRoot], "!root");
```
Comment (Hephyrius) [inline contracts/sidechain/PausableProxyOFT.sol:165]: crit : if we don't dequeue processed hashes, we've got an unlimited mint bug waiting to happen!

File: contracts/sidechain/PausableProxyOFT.sol (hunk 4 above)
Comment (Hephyrius) [inline contracts/sidechain/PausableProxyOFT.sol:166]:
```suggestion
        require(_timestamp.add(queueDelay) >= block.timestamp, "!timestamp");
```

should it not be this way around, or what you're saying is. You gotta execute this withing x amount of time or it'll fail, which would mean using delay in the variable name is misleading

File: contracts/sidechain/PausableProxyOFT.sol (hunk 6 above)
Comment (Hephyrius) [inline contracts/sidechain/PausableProxyOFT.sol:165]:
```suggestion
        require(queue[queueRoot], "!root");
        queue[queueRoot] = false;
```

File: contracts/sidechain/PausableProxyOFT.sol (hunk 4 above)
Comment (phijfry) [reply contracts/sidechain/PausableProxyOFT.sol]: This is the right way already, block.timestamp has to be greater than the delay period, refactored to improve readability though.

**Review by Hephyrius — APPROVED** [review]
lgtm

**Pull request merged**